include/oi/types/st.h

```diff
@@ -16,10 +16,52 @@
 #ifndef OI_TYPES_ST_H
 #define OI_TYPES_ST_H 1
 
-namespace ObjectIntrospection {
-namespace types {
-namespace st {
+/*
+ * Static Types
+ *
+ * OI employs a data segment to transfer information about the probed object to
+ * the debugger. Static Types are used with the `-ftyped-data-segment` feature
+ * to provide a compile time description of the contents of this data segment.
+ *
+ * DataBuffer represents any type with two methods: `void write_byte(uint8_t)`,
+ * which writes a given byte to the buffer; and, `size_t offset()`, which
+ returns
+ * the number of bytes written. The only implementation at present is
```

Comment:
Just a warning: if you're going to say "the only implementation at present" then we'll have to be sure to update this comment if that ever changes. (are you planning to change add more?)

It might be more future-proof to be less specific here, but wording is just a matter of preference so feel free to leave it as-is if you like.